### PR TITLE
Updates for Julia 0.4 and 0.5

### PR DESCRIPTION
**Pull request by dfdx**
* dropped support for Julia 0.3
 * updated code to run on Julia 0.4 (with minor warnings about String in tests)
 * updated code to run on Julia 0.5 (no errors, no warnings)

### CODE REVIEW

File: src/bloom-filter.jl [hunk 1]
```diff
@@ -38,11 +39,12 @@ end
 
 function BloomFilter(mmap_handle::IOStream, capacity::Int, bits_per_elem::Int, k_hashes::Int)
     n_bits = capacity * bits_per_elem
-    mb = mmap_bitarray((n_bits, 1), mmap_handle)
+    # mb = mmap_bitarray((n_bits, 1), mmap_handle)
```
Comment (boydgreenfield) [inline src/bloom-filter.jl]: Generally all looks good to me! Small nit: Mind removing the old `mmap_bitarray` calls?

**Comment by boydgreenfield** [comment]
Generally looks good to me, as long as all the tests are passing. Given the different Julia runtime compatibility, this probably warrants a version bump.

Since this is your change, any preference on `v0.1.2` vs. `v0.2.0` @dfdx? I'm also a bit rusty on my Julia packaging (i.e., I don't v0.4.0 or v0.5.0 installed at the moment). If I tag a release... do I still need to push manually or is the Julia Package Listing pulling updates now?

**Comment by dfdx** [comment]
> Since this is your change, any preference on v0.1.2 vs. v0.2.0
Since we drop support for older versions, the rule of thumb is to increment minor version, i.e. use v0.2.0. 

>  do I still need to push manually or is the Julia Package Listing pulling updates now?

As far as I know Package Listing doesn't pull new versions, so you need to do it manually. IIRC, in Julia < 0.5 it's enough to do:

    Pkg.tag("BloomFilters", :minor)
    Pkg.publish()

or in Julia 0.5:

    Pkg.add("PkgDev")
    PkgDev.tag("BloomFilters", :minor)
    PkgDev.publish()

and open PR to METADATA.jl by clicking the printed link.

**Comment by boydgreenfield** [comment]
OK awesome... I got slammed today but will try to get the latest Julia running this weekend and this merged in. Sorry for the slow ramp and thanks again for the PR @dfdx!

**Comment by dfdx** [comment]
Thanks! I appreciate you still care about the project that you don't use yourself anymore!

**Comment by boydgreenfield** [comment]
Merged, pull request for the update to http://pkg.julialang.org/, and tagged as `v0.2.0` here. Thanks again @dfdx!

**Pull request merged**